easybuild/easyblocks/p/psmpi.py

```diff
@@ -58,9 +124,59 @@ def extra_options(extra_vars=None):
             'mpich_opts': [None, "Optional options to configure MPICH", CUSTOM],
             'threaded': [False, "Enable multithreaded build (which is slower)", CUSTOM],
             'pscom_allin_path': [None, "Enable pscom integration by giving its source path", CUSTOM],
+            'pgo': [False, "Enable profiling guided optimizations", CUSTOM],
+            'mpiexec_cmd': ['srun -n ', "Command to run benchmarks to generate PGO profile. With -n switch", CUSTOM],
         })
         return extra_vars
 
+    def pgo_steps(self):
+        """
+        Set of steps to be performed after the initial installation, if PGO were enabled
+        """
+        self.log.info("Running PGO steps...")
+        # Remove old profiles
+        remove_dir(self.profdir)
+        mkdir(self.profdir)
+
+        # Clean the old build
+        run_cmd('make distclean')
+
+        # Compile and run example to generate profile
+        print_msg("generating PGO profile...")
+        (out, _) = run_cmd('%s 2 hostname' % self.cfg['mpiexec_cmd'])
+        nodes = out.split()
+        if nodes[0] == nodes[1]:
+            raise EasyBuildError("The profile is generated with 1 node! Use 2 nodes to generate a proper profile!")
+
+        write_file('pingpong.c', PINGPONG_PGO_TEST)
+        run_cmd('%s/bin/mpicc pingpong.c -o pingpong' % self.installdir)
```

Comment:
os.path.join(self.installdir, 'bin', 'mpicc') or perhaps the correct ENV var for MPICC